sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1091,6 +1095,19 @@ class Analyzer(
     }
   }
 
+  /**
+   * Resolve [[UnresolvedTableOrView]] by replacing it with [[NotFoundTableOrView]]
+   * if resolution of table or view is not required.
+   *
+   * This rule should run after [[ResolveRelations]] and [[ResolveTables]] are run.
```

Comment:
I remember we shouldn't have a dependency on other rules. I can put these into one rule if the current approach is not desired.